src/main/java/seedu/duke/manager/Manager.java

```diff
@@ -94,28 +89,6 @@ public CommandParser getCommandParser() {
      */
     protected abstract void setSupportedFeatureManagers();
 
-    /**
-     * Checks if the feature is supported by this Manager
-     *
-     * @param featureName Name of the feature to query
-     * @return true if featureName exists under supportedFeatures, else false
-     * @throws NullPointerException is featureName is null
-     */
-    public boolean isSupportedFeature(String featureName) throws NullPointerException {
-        // Sanity check for valid feature name variable
-        if (featureName == null) {
-            throw new NullPointerException("featureName cannot be null!");
-        }
-
-        for (Manager manager : supportedManagers) {
-            String featureKeyword = manager.getFeatureName();
-            if (featureName.equals(featureKeyword)) {
-                return true;
-            }
-        }
-        return false;
-    }
-
```

Comment:
Yes this can be removed and the checking mechanism can be shifted to the main part.